client/js/app/validations/FilterValidations.js

```diff
@@ -46,7 +46,7 @@ module.exports = {
         return true;
       } else if (coercionType === 'Number') {
         return _.isNumber(value);
-      } else if (String(value) === "0") {
+      } else if (coercionType === 'String' || String(value) === "0") {
```

Comment:
You no longer need the `String(value) === "0"` check now that we're allowing empty strings.